app/src/main/java/com/kickstarter/viewmodels/PrivacyViewModel.kt

```diff
@@ -50,91 +51,97 @@ interface PrivacyViewModel {
     }
 
-    class ViewModel(@NonNull val environment: Environment) : ActivityViewModel<PrivacyActivity>(environment), Inputs, Outputs, Errors {
+    class PrivacyViewModel(val environment: Environment) : ViewModel(), Inputs, Outputs, Errors {
         private val optIntoFollowing = PublishSubject.create<Boolean>()
         private val optOutOfFollowing = PublishSubject.create<Boolean>()
         private val userInput = PublishSubject.create<User>()
 
-        private val hideConfirmFollowingOptOutPrompt = BehaviorSubject.create<Void>()
+        private val hideConfirmFollowingOptOutPrompt = BehaviorSubject.create<Unit>()
         private var hidePrivateProfileRow = BehaviorSubject.create<Boolean>()
-        private val showConfirmFollowingOptOutPrompt = BehaviorSubject.create<Void>()
+        private val showConfirmFollowingOptOutPrompt = BehaviorSubject.create<Unit>()
         private val userOutput = BehaviorSubject.create<User>()
-        private val updateSuccess = PublishSubject.create<Void>()
+        private val updateSuccess = PublishSubject.create<Unit>()
 
         private val unableToSavePreferenceError = PublishSubject.create<Throwable>()
 
-        val inputs: PrivacyViewModel.Inputs = this
-        val outputs: PrivacyViewModel.Outputs = this
-        val errors: PrivacyViewModel.Errors = this
+        val inputs: Inputs = this
+        val outputs: Outputs = this
+        val errors: Errors = this
 
-        private val client = requireNotNull(environment.apiClient())
-        private val currentUser = requireNotNull(environment.currentUser())
+        private val client = requireNotNull(environment.apiClientV2())
+        private val currentUser = requireNotNull(environment.currentUserV2())
+        private val disposables = CompositeDisposable()
 
         init {
             this.client.fetchCurrentUser()
                 .retry(2)
-                .compose(Transformers.neverError())
-                .compose(bindToLifecycle())
+                .compose(Transformers.neverErrorV2())
                 .subscribe { this.currentUser.refresh(it) }
+                .addToDisposable(disposables)
 
             val currentUser = this.currentUser.observable()
 
             currentUser
                 .take(1)
-                .filter { it.isNotNull() }
-                .compose(bindToLifecycle())
-                .subscribe { this.userOutput.onNext(it) }
+                .filter { it.isNotNull() && it.getValue().isNotNull() }
+                .map { it.getValue() }
+                .subscribe { this.userOutput.onNext(it!!) }
+                .addToDisposable(disposables)
 
             currentUser
-                .compose(bindToLifecycle())
-                .filter { it.isNotNull() }
-                .map { user -> user.createdProjectsCount().isNonZero() }
-                .subscribe(this.hidePrivateProfileRow)
+                .filter { it.isNotNull() && it.getValue().isNotNull() }
```

Comment:
wouldnt this cause a crash if `it` / `currentUser` is null?

Reply:
no, since the operation happens in order it.isNotNull checks it first, then if it is null the filter fails since its an 'and' check, if it is not null, the second check happens to cover the getValue value

fun <T : Any> T?.isNotNull(): Boolean {
    return this != null
}

Reply:
Ah, what i mean is that im concerned with the first null check. to me, this is implying that the item could be null before hitting that first check, and my understanding is that any null value that's passed down the pipeline is an issue with rxjava 2

Reply:
I believe the concept is called short circuiting.  Basically since its an && it will evaluate the operations in order and if any fail along the way it doesnt continue and just marks the result as false

Reply:
got it! thank you for explaining 👍🏽

Reply:
> Ah, what i mean is that im concerned with the first null check. to me, this is implying that the item could be null before hitting that first check, and my understanding is that any null value that's passed down the pipeline is an issue with rxjava 2

I believe RX2 doesnt like nulls, but as long as you dont emit them you are okay.  I could be wrong about that, but since in this case we are just grabbing an observable im just covering the very odd case its null so that the subsequent it.getValue does not crash

Reply:
i can remove them if you like